internal/binutils/addr2liner.go

```diff
@@ -215,17 +217,22 @@ func (d *addr2Liner) addrInfo(addr uint64) ([]plugin.Frame, error) {
 		return nil, err
 	}
 
-	// Get better name from nm if possible.
+	// Certain versions of addr2line produce incomplete names due to
+	// https://sourceware.org/bugzilla/show_bug.cgi?id=17541. Attempt to replace
+	// the name with a better one from nm.
 	if len(stack) > 0 && d.nm != nil {
 		nm, err := d.nm.addrInfo(addr)
 		if err == nil && len(nm) > 0 {
-			// Last entry in frame list should match since
-			// it is non-inlined. As a simple heuristic,
-			// we only switch to the nm-based name if it
-			// is longer.
+			// Last entry in frame list should match since it is non-inlined. As a
+			// simple heuristic, we only switch to the nm-based name if it is longer
+			// by 2 or more characters. We consider nm names that are longer by 1
+			// character insignificant to avoid replacing foo with _foo on MacOS (for
+			// unknown reasons read2line produces the former and nm produces the
+			// latter on MacOS even though both tools are asked to produce mangled
+			// names.
```

Comment:
Missing close parenthesis to match `(for` on line 229, I think.

Reply:
Done, PTAL.